docs/generate/dataplane-customization.md

```diff
@@ -0,0 +1,96 @@
+# Customization on Data-Plane Client
+
+## Generate Data-Plane Client
+
+Follow [Quickstart](https://aka.ms/azsdk/dpcodegen/java) to generate the Data-Plane Client from OpenAPI specs.
+
+## Change Client Behavior
+
+### Support Azure Active Directory Authentication
+
+See [Authentication in SDK](https://github.com/Azure/autorest/blob/main/docs/generate/authentication.md).
+
+It is advised to update OpenAPI specs with `security: AADToken`. And then re-generate the SDK.
+
+If one does not wish to change the specs, add following configure to README_SPEC.md, and re-generate the SDK.
+```
+security: AADToken
+security-scopes: https://cognitiveservices.azure.com/.default
+```
+Here is an [example](https://github.com/weidongxu-microsoft/azure-sdk-for-java/commit/d30773a776f36e8269c7b2377e1d948fad2f5b82) of the modification of README_SPEC.md and the resulted code changes.
+Note that a new method `credential(TokenCredential tokenCredential)` is generated in builder class.
+
+### Support Customized Authentication
+
+Usually `security: AzureKey` can provide implementation of API key authentication.
+
+For more complex API key authentication, one can customize the builder class.
+
+Here is an [example](https://github.com/weidongxu-microsoft/azure-sdk-for-java/commit/f5337b6fbc5937683e01299a509b505e01d8b7ce) of the modification of builder to support subscription key and API key.
+1. Create `MetricsAdvisorKeys` and `MetricsAdvisorKeyCredential` class in `models` package.
+2. Modify `MetricsAdvisorClientBuilder` to accept `MetricsAdvisorKeyCredential` from user, and then pass the keys in it to HTTP pipeline.
+
+Note that the `Generated` annotation on `createHttpPipeline()` is removed, as we customized this method.
+This ensures that future code generate will not override this customized method.
+
+## Change Operation Behavior
```

Comment:
I don't quite get what 'Change Operation Behavior' means?

Reply:
Quote Laurent

> Required sections are (at the minimum):
> 
> Change operation behavior (new signature, or any wrapping)
> Change client behavior (default pipeline configuration, etc.)
> Add models manually written

For now, use the same phase as the issue for consistency at review. We might change the title later.

Reply:
ok